test/test_utils.py

```diff
@@ -1174,6 +1174,16 @@ def test_build_eager(self):
         for message, value, expected in test_data:
             self.assertEqual(expected, build_eager(value), message)
 
+    def test_duration_condition(self):
```

Comment:
Please move this test to more relevant part o this file, for example after `test_array_should_define_a_set_of_variables`